Cipher.js

```diff
@@ -188,6 +188,46 @@ export class Cipher {
     return JSON.parse(new TextDecoder().decode(data));
   }
 
+  /**
+   * Creates a JOSE Header and encrypted_key for a recipient.
+   *
+   * @see https://tools.ietf.org/html/rfc7516#section-4
+   *
+   * @param {object} options - Options to use.
+   * @param {object} options.recipient - A recipient with a header with a
+   *   kid and alg.
+   * @param {object} options.ephemeralKeyPair - An ephmerealKeyPair.
+   * @param {object} options.cek - A content encryption key.
+   * @param {Function} options.keyResolver - A function that can resolve keys.
+   *
+   * @returns {Promise<object>} A JWE recipient object.
+   */
+  async addRecipient({recipient, ephemeralKeyPair, cek, keyResolver}) {
+    // ensure all recipients use the supported key agreement algorithm
+    const {keyAgreement} = this;
+    const staticPublicKey = await keyResolver({id: recipient.header.kid});
+    // derive KEKs for each recipient
+    const derivedResult = await keyAgreement.kekFromStaticPeer(
+      {ephemeralKeyPair, staticPublicKey});
+    const {kek, epk, apu, apv} = derivedResult;
+    const header = {
+      // contains the key id - kid
+      // contains the algorithm - alg
+      ...recipient.header,
+      // the ephemeralKeyPair
+      epk,
+      // base64 encoded ephemeralKeyPair's publicKey
+      apu,
+      // base64 encoded staticPublicKey's id
+      apv,
+    };
+    return {
+      header,
+      // the cek is wrapped so the recipient can use it to decrypt later
+      encrypted_key: await kek.wrapKey({unwrappedKey: cek})
+    };
```

Comment:
The above will strip out other custom fields from `recipient` that we must pass on. Please make sure those don't get removed.